### PR TITLE
Add codecov and black (melodic-devel)

### PR DESCRIPTION
**Pull request by tylerjw**
This also fixes issues with ccache and adds the uploading of test failures.

### CODE REVIEW

File: .github/workflows/industrial_ci_action.yml [hunk 1]
```diff
@@ -10,15 +10,22 @@ jobs:
...
+            CMAKE_ARGS: "-DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_FLAGS='--coverage' -DCMAKE_CXX_FLAGS='--coverage'",
+            AFTER_RUN_TARGET_TEST: './.ci.prepare_codecov',
+            ADDITIONAL_DEBS: 'lcov'
+          }
+          - {
+            ROS_DISTRO: kinetic, ROS_REPO: main, CCOV_UPLOAD: false, 
+            BEFORE_BUILD_TARGET_WORKSPACE: './.ci.mongo_source_install', 
+            ROSDEP_SKIP_KEYS: 'libmongoclient-dev',
+            ADDITIONAL_DEBS: 'libssl-dev scons git'
+          }
```
Comment (rhaschke) [inline .github/workflows/industrial_ci_action.yml]: Please reformat as follows (each dict entry should have its own line):
```yaml
        env:
          - {
              ROS_DISTRO: melodic,
              ROS_REPO: main,
              CCOV_UPLOAD: true,
              CATKIN_LINT: pedatntic,
              CMAKE_ARGS: "-DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_FLAGS='--coverage' -DCMAKE_CXX_FLAGS='--coverage'",
              AFTER_RUN_TARGET_TEST: "./.ci.prepare_codecov",
              ADDITIONAL_DEBS: "lcov",
            }
          - {
              ROS_DISTRO: kinetic,
              ROS_REPO: main,
              CCOV_UPLOAD: false,
              BEFORE_BUILD_TARGET_WORKSPACE: "./.ci.mongo_source_install",
              ROSDEP_SKIP_KEYS: "libmongoclient-dev",
              ADDITIONAL_DEBS: "libssl-dev scons git",
            }
```

**Review by rhaschke — APPROVED** [review]
One minor nit. Otherwise lgtm.

File: .github/workflows/industrial_ci_action.yml (hunk 1 above)
Comment (tylerjw) [reply .github/workflows/industrial_ci_action.yml]: I did this and brought in the change of the cache prefix.  I really like how that simplifies the cache keys.

**Comment by tylerjw** [comment]
One more thing, I removed the large codecov graph from the readme and left the codecov badge only.

**Review by rhaschke — APPROVED** [review]
We should really consider [setting up our own repo of composite actions](https://docs.github.com/en/actions/creating-actions/creating-a-composite-run-steps-action) instead of adding the same boilerplate files everywhere. This is a nightmare to update / keep in sync in future.

**Pull request merged**